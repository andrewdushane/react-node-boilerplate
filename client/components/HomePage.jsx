import React, { PropTypes } from 'react';
import { connect } from 'react-redux';

import { exampleRequest, clearMessage } from '../actions';

export const Home = ({ exampleRequest: request, clearMessage: clear, message }) => (
  <div>
    <header>
      Hello, World
    </header>
    <section>
      <button onClick={request}>Make an API request</button>
    </section>
    {message &&
      <section>
        <strong>Message:</strong> { message }
        <div>
          <button onClick={clear}>Clear Message</button>
        </div>
      </section>
    }
  </div>
);

Home.propTypes = {
  exampleRequest: PropTypes.func.isRequired,
  clearMessage: PropTypes.func.isRequired,
  message: PropTypes.string,
};

const mapStateToProps = ({ example }) => ({
  message: example.message,
});

const dispatchProps = {
  exampleRequest,
  clearMessage,
};

export default connect(mapStateToProps, dispatchProps)(Home);
