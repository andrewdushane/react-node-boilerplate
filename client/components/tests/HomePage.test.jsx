import React from 'react';
import renderer from 'react-test-renderer';
import { Home } from '../HomePage';

const props = {
  exampleRequest: jest.fn(),
  clearMessage: jest.fn(),
};

test('Home renders as expected', () => {
  const component = renderer.create(
    <Home {...props} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
