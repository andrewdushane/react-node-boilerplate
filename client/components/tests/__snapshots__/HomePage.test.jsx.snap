exports[`test Home renders as expected 1`] = `
<div>
  <header>
    Hello, World
  </header>
  <section>
    <button
      onClick={[Function]}>
      Make an API request
    </button>
  </section>
</div>
`;
